src/Streamly/Internal/Data/Stream/IsStream/Transform.hs

```diff
@@ -991,8 +993,16 @@ takeWhileAround = undefined -- fromStreamD $ D.takeWhileAround n $ toStreamD m
 -- /Internal/
 --
 {-# INLINE takeByTime #-}
-takeByTime ::(MonadIO m, IsStream t, TimeUnit64 d) => d -> t m a -> t m a
-takeByTime d = fromStreamD . D.takeByTime d . toStreamD
+takeByTime ::
+       (MonadAsync m, IsStream t, TimeUnit64 d, Functor (t m))
+    => d -> t m a -> t m a
+takeByTime duration = catMaybes . Par.parallelMin timeStream . fmap Just
```

Comment:
we can flip the arguments and use `parallelFst` since we know one of the streams is infinite, parallelFst may be a bit more efficient as it only checks one stream, it may be negligible though.

Reply:
Behavior change: The earlier implementation would terminate immediately when the time is over. This one would depend on the granularity of the relTimes clock. So we need to let the user specify the granularity as well to control it.